mesonbuild/linkers.py

```diff
@@ -137,7 +137,7 @@ def __init__(self, exelist: typing.List[str], machine: str):
 class ArLinker(StaticLinker):
 
     def __init__(self, exelist: typing.List[str]):
-        super().__init__(exelist)
+        StaticLinker.__init__(self, exelist)
```

Comment:
Why does it complain about this particular case? This is the one case where `super` is exactly what one should do AFAICT.

Reply:
I think it complains because we were mixing `super()` and `XXX.__init__()` style. Specifically, [here](https://github.com/mesonbuild/meson/blob/master/mesonbuild/linkers.py#L119) is the explicit syntax required. This is not allowed/recommended from what I can tell from the [LGTM docs](https://lgtm.com/rules/5990072).

So, using `super()` everywhere would require even more refactoring (ping @dcbaker).

Reply:
Ah, right, probably because some class that inherits from this uses non-super. This is fine as such, because it improves things. Should all of this be squashed to one commit or should the one extra be merged into some other commit?

Reply:
I will manually squash the extra commit once the CI is green. Not squashing all the commits would make the inevitable `git bisect` due to unforseen refactoring consequences less of a pain :)

Reply:
That will retrigger the CI again. I can do the merge manually if you tell which commit it should be merged into.